tests: add test to verify static libraries are expected

kevinburke opened the pull request:
This is an subtle way for things to break, it's easy to accidentally
break it, and the check for it is not _that_ complicated, so we can
automate it.

jsha reviewed (changes requested):
Thanks for working on this!

kevinburke commented:
Moved the logic to a script, and moved test.sh into `tests` (maybe `tests/bin` would be better?) since we now have two test scripts and I didn't want to clutter the top level.

kevinburke commented on .github/workflows/test.yaml:
```diff
@@ -60,6 +60,33 @@ jobs:
+          if [[ "${libs}" != '-lgcc_s -lutil -lrt -lpthread -lm -ldl -lc' ]]; then
+            echo "got unexpected list of static libraries, fix or update README: ${libs}"
+            exit 1
+          fi
```
I made this change, but note it makes it more difficult / obscures the contents that are actually being reported now, since they're grepped out.

jsha reviewed (approved):
Thanks!
